crates/api/src/externals.rs

```diff
@@ -56,7 +56,7 @@ impl Extern {
         }
     }
 
-    pub(crate) fn get_wasmtime_export(&mut self) -> wasmtime_runtime::Export {
+    pub(crate) fn get_wasmtime_export(&self) -> wasmtime_runtime::Export {
```

Comment:
A `wasmtime_runtime::Export` contains raw pointer to mutable data (the memory, global, or table storage). Is it logical to think of that as part of `self`'s state? If so, `self` should remain mutable here.

Reply:
Oh hm I'm not actually sure, the input to one function was `&[Extern]` and I saw that this didn't need the `&mut self` so I figured this could be removed.

If it's internally got self-pointers though then I think a better fix might be to have a lifetime parameter on `Export` to connect it to `&self`. The `&mut`-ness won't actually matter in terms of connections here, especially because `Export` is cloneable you can get a `&mut `by simply cloning it as well.